test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -10854,6 +10854,228 @@ public async Task RestoreNetCore_WhenPackageNamespacesConfiguredAndNoMatchingSou
+            projectA.AddPackageToAllFrameworks(packageOpenSourceContosoMvc);
+
+            solution.Projects.Add(projectA);
+            solution.Create(pathContext.SolutionRoot);
+
+            // SimpleTestPathContext adds a NuGet.Config with a repositoryPath,
+            // so we go ahead and replace that config before running MSBuild.
+            var configAPath = Path.Combine(Path.GetDirectoryName(projectA.ProjectPath), "NuGet.Config");
+            var configText =
+$@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""PublicRepository"" value=""{opensourceRepositoryPath}"" />
+    <add key=""PrivateRepository"" value=""{privateRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""PublicRepository""> 
+            <namespace id=""Contoso.Opensource.*"" />
+        </packageSource>
+        <packageSource key=""PrivateRepository"">
+            <namespace id=""Contoso.MVC.*"" /> <!--Contoso.MVC.ASP package exist in both repository but it'll restore from this one -->
+        </packageSource>
+    </packageNamespaces>
+</configuration>";
+            using (var writer = new StreamWriter(configAPath))
+            {
+                writer.Write(configText);
+            }
+
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                opensourceRepositoryPath,
+                packageOpenSourceA);
+
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                opensourceRepositoryPath,
+                packageOpenSourceContosoMvc);
+
+
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                privateRepositoryPath,
+                packageContosoMvcReal);
+
+            var packagePath = Path.Combine(pathContext.WorkingDirectory, "packages");
+
+            string[] args = new string[]
+                {
+                        "-OutputDirectory",
+                        "packages"
+                };
+
+            // Act
+            var r = Util.Restore(pathContext, projectA.ProjectPath, expectedExitCode: 0, args);
+
+            // Assert
+            // If we pass source then log include actual path to repository instead of repository name.
+            Assert.Contains($"Package namespace matches found for package ID 'Contoso.MVC.ASP' are: 'PrivateRepository'", r.Output);
+            Assert.Contains($"Package namespace matches found for package ID 'Contoso.Opensource.A' are: 'PublicRepository'", r.Output);
+            var contosoRestorePath = Path.Combine(packagePath, packageContosoMvcReal.Id.ToString(), packageContosoMvcReal.Version.ToString(), packageContosoMvcReal.ToString() + ".nupkg");
+            using (var nupkgReader = new PackageArchiveReader(contosoRestorePath))
+            {
+                var allFiles = nupkgReader.GetFiles().ToList();
```

Comment:
PR projects should validate the nupkg metadata instead. cc @kartheekp-ms

Reply:
@nkolev92 
Addressed.